README.md

```diff
@@ -2618,4 +2624,121 @@ React is a JavaScript library for building user interfaces. It takes the backend
+
+#### Key HTTP Methods
+
+- `GET` - Client asks web server for data.
+- `POST` - Client sends data to web server.
+- `PUT` - Client requests that web server updates data (client sends entirety of data).
+- `PATCH` - Client requests that web server updates specific data (client only sends relevant data).
+- `DELETE` - Client requests that web server deletes data.
+
+Web servers are not required to follow these methods, and can use methods (e.g. `PUT`) to action something else.
+
+When sent from the client, methods such as `POST`, `PUT` and `PATCH` always consist of a header and a body (as the client needs to send the web server the data it wants to add or update).
+
+The `GET` method does not need a body when sent from the client as additional information can be included in the Uniform Resource Locator (URL).
+
+The body is sometimes referred to as the payload. The responses from the web server will always have a body.
+
+The sender will specify how many bytes the body consists of. Generally each character is one byte, although some characters e.g. Chinese characters could be more than one byte. If the body received contains a smaller or larger number of bytes, then the receiver knows that the content could have been corrupted.
+
+### HTTP Status Codes
+
+HTTP status codes are all 3 digits long. More commonly known ones are `200` - 'OK', `202` - 'Accepted' and `404` - 'Data Not Found'.
+
+When they begin with the following numbers, status codes mean:
+
+- `1..` = hang on
+- `2..` = ok
+- `3..` = redirecting
+- `4..` = user messed up
+- `5..` = server messed up
+
+### Uniform Resource Locator (URL)
+
+The client to web server communications are enabled by web browsers. Web browsers are software applications with Graphical User Interfaces (GUIs) that take the request made by the person/computer and retrieve it from a web server using a Uniform Resource Locator (URL).
+
+URLs are signposts to web servers and usually consist of domains, resources and query parameters.
+
+The URL `https://www.amazon.com` is in the form of HTTPS and consists of three domains. `www` is a sub-domain of the `amazon` domain and `com` is the top level domain. Large companies such as Amazon will purchase all sub-domains and top level domains of their domain (e.g. `amazon`) so they can re-direct all potential customers to their main site.
+
+The following is part of a URL of a Google search for cat videos:
+
+`https://www.google.com/search?client=ubuntu&channel=fs&ei=0Bq7XNmKEsj8kwXDyaC4Bg&q=cat+videos`
+
+It is in the form of HTTPS, with a sub-domain of `www`, a domain of `google` and a top level domain of `com`. This URL also has a resource and query parameters. The resource is `search`. Resources consist of nouns and/or IDs and always follow on straight after the domains. The query parameters begin with `?` and are added to with `&`.
```

Comment:
Can you also explain the key-value relationship that is created with the equals sign?